ntbs-service/DataMigration/NotificationMapper.cs

```diff
@@ -7,98 +7,84 @@
 
-                var results = await connection.QueryAsync(NotificationsQuery, new
-                {
-                    NotificationId = $"{notificationId}"
-                });
+            return GetGroupedResultsAsNotification(notificationsRaw, notificationSitesRaw);
+        }
+
+        public async Task<List<List<Notification>>> GetNotificationsGroupedByPatient(List<string> notificationIds)
+        {
+            var notificationsRaw = await _migrationRepository.GetNotificationsById(notificationIds);
+            var legacyIds = notificationsRaw.Select(x => x.OldNotificationId).Cast<string>();
+            var notificationSitesRaw = await _migrationRepository.GetNotificationSites(legacyIds);
 
-                var notifications = results.Select(AsNotification).ToList();
+            return GetGroupedResultsAsNotification(notificationsRaw, notificationSitesRaw);
+        }
 
-                var idsForInClause = string.Join(",", notifications.Select(x => $@"'{x.LegacyId}'"));
-                var notificationSites = await connection.QueryAsync(string.Format(NotificationSitesQuery, idsForInClause));
-                var groupingByLegacyId = notificationSites.GroupBy(x => x.OldNotificationId);
-                foreach (var group in groupingByLegacyId)
+        private List<List<Notification>> GetGroupedResultsAsNotification(IEnumerable<dynamic> notificationsRaw, IEnumerable<dynamic> notificationSitesRaw)
+        {
+            var groupedResults = new List<List<Notification>>();
+            var notificationGroups = notificationsRaw.GroupBy(x => x.GroupId);
+            var notificationSiteGroups = notificationSitesRaw.GroupBy(x => x.OldNotificationId);
+            
+            var notificationSiteDictionary = new Dictionary<string, List<NotificationSite>>();
+            foreach (var notificationSiteGroup in notificationSiteGroups)
+            {
+                var notificationSites = notificationSiteGroup.Select(AsNotificationSite).Where(x => x != null);
+                if (notificationSites.Count() > 0)
                 {
-                    notifications.FirstOrDefault(x => x.LegacyId == group.Key).NotificationSites = group.Select(AsNotificationSite).ToList();
+                    notificationSiteDictionary.Add(notificationSiteGroup.Key, notificationSites.ToList());
                 }
+            }
 
-                return notifications;
+            foreach (var notificationGroup in notificationGroups)
+            {
+                var notifications = notificationGroup.Select(AsNotification).ToList();
+                notifications.ForEach(x => {
+                    if (notificationSiteDictionary.ContainsKey(x.LegacyId) && notificationSiteDictionary[x.LegacyId] != null)
+                    {
+                        x.NotificationSites = notificationSiteDictionary[x.LegacyId];
+                    }
+                });
+                
+                groupedResults.Add(notifications);

```

Comment:
Most of this makes perfect sense, but I think we can simplify it a tiny bit by attaching the sites first, before we do results. let's spend a few min pairing on this in person?